### PR TITLE
Fix/double

### PR DESCRIPTION
**Pull request by zflamig**
Don't do a status check
Don't double request on GET
Be more defensive from invalid returns

### CODE REVIEW

File: indexclient/client.py [hunk 1]
```diff
@@ -173,10 +175,10 @@ def to_json(self, include_rev=True):
         json["did"] = self.did
         return json
 
-    def refresh(self):
+    def refresh(self, json=None):
```
Comment (philloooo) [inline indexclient/client.py]: the added parameter breaks the purpose of this function, just don't call refresh for initialization?

Comment (zflamig) [reply indexclient/client.py]: How does it break it?? The json value isn't cached. I don't want to remove refresh from init some code may be relying on that behavior?

Comment (philloooo) [reply indexclient/client.py]: aren't you just not calling server to get latest json if a json is passed? basically you don't refresh if a json is passed?

File: indexclient/client.py [hunk 2]
```diff
@@ -40,13 +38,17 @@ def check_status(self):
...
         try:
-            self._get("index", did)
+            response = self._get("index", did)
         except requests.HTTPError as e:
             if e.response.status_code == 404:
                 return None
             else:
-                raise e
-        return Document(self, did)
+                print("Exception!!" + str(e.reason))
```
Comment (philloooo) [inline indexclient/client.py]: I think it's cleaner to not overload `None` to mean both not found and something wrong happened?

File: indexclient/client.py (hunk 1 above)
Comment (zflamig) [reply indexclient/client.py]: If you pass a json it loads it from that json. Its why we still need this function to translate from json -> internal structure. The function could be renamed from refresh to load, but that might break people so its not worth it.

Comment (philloooo) [reply indexclient/client.py]: nothing else is using it as it serves more like a private method, let's rename it and update docstring to minimize confusion for future?

File: indexclient/client.py [hunk 3]
```diff
@@ -38,16 +37,32 @@ def check_status(self):
         resp = requests.get(self.url + '/index')
         handle_error(resp)
 
+    def global_get(self, did, no_dist=False):
+        """Return a document object corresponding to a single did from the global endpoint"""
```
Comment (philloooo) [inline indexclient/client.py]: what's no_dist? docstring?

File: indexclient/client.py [hunk 4]
```diff
@@ -197,7 +213,7 @@ def patch(self):
                          headers={"content-type": "application/json"},
                          auth=self.client.auth,
                          data=json_dumps(self._doc_for_update()))
-        self.refresh()  # to sync new rev from server
+        self.load()  # to sync new rev from server
```
Comment (philloooo) [inline indexclient/client.py]: _load ?

**Pull request merged**